File: api/v1/Item.js
import Request from "@/api/request.js"
let request = new Request().http


/***
 * @description: 获取单品
 * @param . pagenum:偏移量, pagesize:数量,cid:分类id
 */
export function getItems(pagenum, pagesize, cid) {
  return request({
    url: "/api/v1/items",
    method: "GET",
    data: {
      pagenum,
      pagesize,
      cid
    }
  })
}

/***
 * @description: 增加单品
 * @param . name:单品名, description:单品描述, unit:单位, category_id:分类id, picture:图片, price:价格, status:状态
 */
export function addItem(data) {
  return request({
    url: "/api/v1/item/add",
    method: "POST",
    data
  })
}

File: api/v1/item.go
package v1

import (
	"DineEasy/model"
	"DineEasy/utils/errmsg"
	"DineEasy/utils/tool"
	"github.com/gin-gonic/gin"
	"strconv"
)

// AddItem 添加单品
func AddItem(c *gin.Context) (int, any) {
	var data model.Item
	err = c.ShouldBindJSON(&data)
	if err != nil {
		return errmsg.ERROR_BIND, err
	}
	code, itemID := model.AddItem(&data)
	return code, gin.H{"item_id": itemID}
}

// GetItems 获取单品列表
func GetItems(c *gin.Context) (int, any) {
	pageNum, pageSize := tool.PageTool(c)
	cid, _ := strconv.Atoi(c.Query("cid"))
	if cid == 0 {
		cid = 1
	}
	code, data, count := model.GetItems(pageNum, pageSize, cid)
	return code, gin.H{"data": data, "count": count}
}
